packages/react/src/styled/__tests__/types.test.js.flow

```diff
@@ -1,30 +1,70 @@
 // @flow strict-local
-import React, { type Node } from 'react';
-import { styled, type CssObject, type CSSProps } from '@compiled/react';
+import React, { type Node, type Element } from 'react';
+import { styled, type CssType } from '@compiled/react';
 
 // Tagged template expression
 () => {
-  const StyledTaggedTemplateExpression = styled.div`
+  const StyledComponent = styled.div`
     font-size: 12px;
   `;
-  return <StyledTaggedTemplateExpression>red text</StyledTaggedTemplateExpression>;
+  return <StyledComponent>red text</StyledComponent>;
+};
+
+// Tagged template expression with conditional
+() => {
+  const StyledComponent = styled.div`
+    font-size: ${({ big }) => (big ? big : 12)}px;
+  `;
+  return <StyledComponent big>red text</StyledComponent>;
 };
 
 // Object call expression
 () => {
-  const StyledTaggedTemplateExpression = styled.div({ fontSize: 12 });
-  return <StyledTaggedTemplateExpression>red text</StyledTaggedTemplateExpression>;
+  const StyledComponent = styled.div<{ big: number, children: Node }>({
+    fontSize: (props: { big: number, children: Node }): number => (props.big == 5 ? 50 : 10),
+  });
+
+  return <StyledComponent big={5}>red text</StyledComponent>;
+};
+
+// Object call expression with no props
+() => {
+  const StyledComponent = styled.div({
+    wordBreak: 'break-word',
+    position: 'relative',
+    left: `${2 - 2}px`,
+  });
+
+  return <StyledComponent>red text</StyledComponent>;
+};
+
+// Object call expression with conditional
+() => {
+  const StyledComponent = styled.div({
+    fontSize: ({ big }) => `${big ? 50 : 12}px`,
+    width: ({ big }) => (big ? 500 : 120),
```

Comment:
I implemented something similar but wasn't sure exactly how this would work. Does it mean:
```
...(({ big }) => big && { height: '100px' })
```?

Reply:
Nah, it would be like this:

```
styled.div<{ big: number, children: Node }>(
  {
    fontSize: ({ big }) => `${big ? 50 : 12}px`,
    width: ({ big }) => (big ? 500 : 120),
  },
  (({ big }) => big && { height: '100px' }),
);
```